crates/shared/src/price_estimation.rs

```diff
@@ -457,53 +457,7 @@ pub type PriceEstimateResult = Result<Estimate, PriceEstimationError>;
 pub trait PriceEstimating: Send + Sync + 'static {
     // The '_ lifetime in the return value is the same as 'a but we need to write it
     // as underscore because of a mockall limitation.
-
-    /// Returns one result for each query in arbitrary order. The usize is the
-    /// index into the queries slice.
-    fn estimates<'a>(&'a self, queries: &'a [Query])
-        -> BoxStream<'_, (usize, PriceEstimateResult)>;
-}
-
-/// Use a PriceEstimating with a single query.
-pub async fn single_estimate(
-    estimator: &dyn PriceEstimating,
-    query: &Query,
-) -> PriceEstimateResult {
-    estimator
-        .estimates(std::slice::from_ref(query))
-        .next()
-        .await
-        .unwrap()
-        .1
-}
-
-/// Use a streaming PriceEstimating with the old Vec based interface.
-pub async fn vec_estimates(
-    estimator: &dyn PriceEstimating,
-    queries: &[Query],
-) -> Vec<PriceEstimateResult> {
-    let mut results = vec![None; queries.len()];
-    let mut stream = estimator.estimates(queries);
-    while let Some((index, result)) = stream.next().await {
-        results[index] = Some(result);
-    }
-    let results = results.into_iter().flatten().collect::<Vec<_>>();
-    // Check that every query has a result.
-    debug_assert_eq!(results.len(), queries.len());
-    results
-}
-
-/// Convert an old Vec based PriceEstimating implementation to a stream.
-pub fn old_estimator_to_stream<'a, IntoIter>(
-    estimator: impl Future<Output = IntoIter> + Send + 'a,
-) -> BoxStream<'a, (usize, PriceEstimateResult)>
-where
-    IntoIter: IntoIterator<Item = PriceEstimateResult> + Send + 'a,
-    IntoIter::IntoIter: Send + 'a,
-{
-    futures::stream::once(estimator)
-        .flat_map(|iter| futures::stream::iter(iter.into_iter().enumerate()))
-        .boxed()
+    fn estimate<'a>(&'a self, query: &'a Query) -> BoxFuture<'_, PriceEstimateResult>;
```

Comment:
nit: while we are on it, can we pass an owned Query here? This will make lifetime considerations much simpler (otherwise whatever call estimate needs to have a lifetime that is at least `'a` which can de difficult if we are just passing the Future to some outer consumer.

Reply:
One argument for keeping `Query` is that they include `Vec<u8>`s of hook calldata. In general, I do think we should remove the lifetime association between `&query` and the box future (and keep it only tied to the lifetime of `self`) if it makes code easier (pre-mature optimizations and all). We can also just change that if needed at some point (since the current lifetime works now and, AFAIU, doesn't cause any issues with the existing code).

Reply:
Alternatively we can also use `Arc<Query>` to keep ownership semantics while keeping copies cheap.